solr/core/src/java/org/apache/solr/update/TransactionLog.java

```diff
@@ -247,6 +248,9 @@ protected TransactionLog(
         } catch (Exception e) {
           log.error("Error closing tlog file (after error opening)", e);
         }
+        if (!openExisting) {
```

Comment:
Main point of this fix: delete the transaction log file in case of exception. Otherwise the "id" field in UpdateLog will always point to this file and next TransactionLog constructor calls will always throw the "New transaction log already exists" exception, blocking all updates.